relationships.go

```diff
@@ -0,0 +1,66 @@
+package dbl
+
+// Relationships help to store the relationships for an Entry
+type Relationships []Relationship
+
+// Append will append a set of relationship IDs for a given relationship
+func (r *Relationships) Append(relationshipIDs ...string) {
+	*r = append(*r, relationshipIDs)
+}
+
+// Relationship help to store the IDs for an Entry's relationship
+type Relationship []string
+
+// Has will note if a relationship has a particular relationship ID
+func (r Relationship) Has(relationshipID string) (has bool) {
+	for _, id := range r {
+		if id == relationshipID {
+			return true
+		}
+	}
+
+	return false
+}
+
+func (r Relationship) delta(old Relationship, onAdd, onRemove RelationshipFn) (err error) {
+	if err = r.addNew(old, onAdd); err != nil {
+		return
+	}
+
+	if err = r.removeOld(old, onRemove); err != nil {
+		return
+	}
```

Comment:
If the two operations are not independent of each other, I would do them concurrently. 

Current time complexity is `( N + M )`, we can make it `Max(N,M)` by concurrency. As, number of operations are known here, so I would make use of `wait group`.

Reply:
Then we are adding 4kb worth of goroutines every time. If our lists were large (1000+), this would totally be an option I would want to approach. Generally these slices are 3-6 in length.

Reply:
Agreed, num relationships won't scale nearly as quickly as overall entry count.